lib/finalize-manifest.js

```diff
@@ -188,7 +194,8 @@ function tarballedProps (pkg, spec, opts) {
           _resolved: (mani && mani._resolved) ||
           (pkg && pkg._resolved) ||
           spec.fetchSpec,
-          _integrity: hash && hash.toString()
+          _integrity: needsIntegrity && hash && hash.toString(),
```

Comment:
This should probably be `hash.sha512.toString()`, since we don't currently put sha1s into the `integrity` field.